### ISSUE TITLE
Trilinos with MKL/BLAS/LAPACK using CMake within VS 2019, MS C++ and Intel Fortran compilers

### ISSUE THREAD

**Issue by ajkerman** [trilinos/Trilinos#6753]
I have been trying unsuccessfully for several weeks to build and link Trilinos Anasazi using the built-in CMake interface of Visual Studio 2019 and the VS19 CMake generator, using MKL, BLAS, and LAPACK TPLs. The number of different failed attempts is so great that i can't usefully list them all. I have scaled back my attempts to only Epetra and Teuchos for now, still with no success.

Could anyone please narrow down the possibilities for:
-which toolset in MSVC to use (i.e., msvc_x64_x64, msvc_x86_x64, etc...)
-which Intel Fortran compiler (x64 or x64_x86)
-which MKL, BLAS, and LAPACK LIBRARY_NAMES to use (I have tried the x86 ones, the x64_x86 and the x64_x64, based on the Intel MKL link line advisor)
-static or shared libraries

selected errors i have encountered:
during CMake, i find (with one exception that i cannot reproduce)
-unable to find dggsvd3 when processing Epetra 
-TEUCHOS_BLASFLOAT and COMPLEX_BLAS fail when processing Teuchos

Even when i can get Trilinos to build and install, i always still get link errors when i try to compile any C++ code, and lots of warnings about type 64 bit conversions.

I have the strong suspicion that even when i delete the CMakeFiles folder and the CMakeLists.txt file in the Trilinos build directory, that there is still some memory of previous setting being inherited from somewhere, because there seems to still be some hysteresis in the outcome (that is, i return as far as i can see to the same settings and get different results)

I apologize in advance for my ignorance about much of this. Trying to learn by doing.
Please help!

**Comment by mhoemmen** [comment]
@trilinos/anasazi @trilinos/framework

**Comment by ajkerman** [comment]
I found that switching to the Intel C++ compiler (19.1) has solved some of the problems. Now I can get success for the BLAS-related tests such as HAVE_TEUCHOS_BLASFLOAT, COMPLEX_BLAS_WORKS; 

However, when i try to compile a program and link it to Trilinos, i get a bunch of link errors that all look like:

error LNK2019: unresolved external symbol "public: __cdecl Epetra_SerialComm::Epetra_SerialComm(void)" (??0Epetra_SerialComm@@QEAA@XZ) referenced in function main [C:\Users\AN18856\source\repos\Trilinos_test\out\build\x64-Debug\JJcircuitSim_CM.vcxproj]

They all share the reference to the _cdecl interface.

I had built Trilinos using the Intel MKL libraries mkl_intel_lp64, mkl_sequential, mkl_core; Since _cdecl appears to refer to a 32 bit interface, i tried rebuilding trilinos with the corresponding 32-bit Intel libraries: mkl_intel_c, mkl_sequential, mkl_core. However, this cause the Trilinos build to fail all of the BLAS related tests again.
